prometheus_juju_backup_all_exporter/__main__.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+import logging
+
+from .collector import BackupStatusCollector
+from .config import DEFAULT_CONFIG, Config
+from .exporter import Exporter
+
+root_logger = logging.getLogger()
+
+
+def parse_command_line():
+    """Command line parser.
+
+    Parse command line arguments and return the arguments.
+
+    Returns:
+        args: Command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        prog=__package__,
+        description=__doc__,
+    )
+    parser.add_argument(
+        "-c", "--config", help="Set configuration file.", default="", type=str
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    args = parse_command_line()
+    config = Config.load_config(config_file=args.config or DEFAULT_CONFIG)
+    root_logger.setLevel(logging.getLevelName(config.level))
```

Comment:
I think it won't work if the value of config.level is defined by us 
It is a Enum (class LogLevel(str, Enum)) , isn't it?

logger.setLevel(logging.DEBUG)
logger.setLevel(logging.getLevelName(logging.DEBUG))
logger.setLevel("DEBUG")

These are workable.

Reply:
It's both `str` and `Enum`

```python3
from enum import Enum
from pydantic import BaseModel

class LogLevel(str, Enum):
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"
    CRITICAL = "CRITICAL"


class Config(BaseModel):
    port: int = 10000
    level: LogLevel = LogLevel.DEBUG

c = Config(level = "DEBUG")
print(c, isinstance(c.level, str), isinstance(c.level, Enum))
# Return: port=10000 level=<LogLevel.DEBUG: 'DEBUG'> True True

try:
    c = Config(level = "RANDOM")
except Exception as e:
    print(e)
    # 1 validation error for Config
    #level 
    #value is not a valid enumeration member; permitted: 'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'    type=type_error.enum; enum_values=[<LogLevel.DEBUG: 'DEBUG'>, <LogLevel.INFO: 'INFO'>, #<LogLevel.WARNING: 'WARNING'>, <LogLevel.ERROR: 'ERROR'>, <LogLevel.CRITICAL: 'CRITICAL'>])

```